components/playlist/browser/resources/utils/urlFixer.ts

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { PlaylistItem } from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m.js'
+import { Url } from 'gen/url/mojom/url.mojom.m'
+
+export function fixUpThumbnailURL (item: PlaylistItem) {
+  if (!item.thumbnailPath?.url) {
+    return
+  }
+
+  if (
+    item.thumbnailPath.url.startsWith('http://') ||
```

Comment:
Is it common for the thumbnails to be hosted as an http resource? It would be ideal to keep this as only `https://`

Reply:
Not common but I think we need both, beacuse we can't sure what it could be in the wild 😄